### DISCUSSION TITLE
Windows executable versions with openBLAS (not AVX)

### DISCUSSION THREAD

**Discussion by xyont** [TLCFEM/suanPan#120] [General]
dear Developer,

many thanks for such a great software application with GPL distributions.

Windows OS commonly used by many student, but the processor and generation may vary. is it possible to use another library solver? even only to be familiar and exploring SuanPan capabilities to solved a small or moderate models.

to my understanding FE most depend on memory, so i suggest to distribute executable for windows without AVX dependencies, hope openBlas is not hard to integrated.

best regards,

**Comment by TLCFEM** [comment]
AVX was introduced in sandy bridge, which was already 11 years ago. The executables can run in CI/CD VMs, so I think it is probably safe to enable it.

AVX support is not tied to any specific linear algebra driver, it is enabled by default when compiling the application, and it can be turned off by assigning `-DUSE_AVX=OFF` in CMake.

For openBLAS or MKL, there is no requirement at all. The [current](https://github.com/TLCFEM/prebuilds/blob/master/.github/workflows/v321-openblas-win.yml) openBLAS is configued using

```
DYNAMIC_ARCH = 1
BINARY=64
USE_THREAD = 1
USE_OPENMP = 1
NUM_THREADS = 20
NO_CBLAS = 1
NO_LAPACKE = 1
GEMM_MULTITHREAD_THRESHOLD = 64
```

If you are able to compile the application by yourself, you can build your own version of openBLAS, drop both `libopenblas.lib` and `libopenblas.dll` in `Libs/win`, then compile the main program with `-DUSE_AVX=OFF`. Check [CI/CD configuration](https://github.com/TLCFEM/suanPan/blob/861855694c8007d1dc18b13a560a76ccc9cd2e8f/.github/workflows/master-vtk.yml#L78) as a reference.

If you are looking for an openBLAS version, it is already [there](https://github.com/TLCFEM/suanPan/releases/download/suanPan-v2.8/suanPan-win-openblas-vtk.7z), but with AVX enabled.

Anyway, it is not a bad idea to ship non-AVX versions. I may do it in the next release.

**Comment by TLCFEM** [comment]
And yes, solving matrices is likely to be memory bandwidth bounded.

**Comment by xyont** [comment]
right, AVX is an old. however some processor type did not support e.g celeron. 

another FE software still usable on my second pc with 16GB memory, it's a CalculiX with PaStiX solver running multipart contact, plasticity and large deformation just below an hours.

many thanks to consider additional executable distributions without AVX activated, given wide range opportunity to many user or student to learn and use SuanPan.

**Reply by TLCFEM** [reply]
I've made a release with non-AVX binaries, see if it meets your need.
